Detectors/PHOS/reconstruction/src/Clusterer.cxx

```diff
@@ -113,23 +169,265 @@ void Clusterer::MakeClusters(const std::vector<Digit>* digits, std::vector<Clust
     } // loop over cluster
   }   // energy theshold
 }
+//__________________________________________________________________________
+void Clusterer::makeUnfoldings(const std::vector<Digit>* digits)
+{
+  //Split cluster if several local maxima are found
+
+  int* maxAt = new int[o2::phos::PHOSSimParams::Instance().mNLMMax]; // NLMMax:Maximal number of local maxima
+  float* maxAtEnergy = new float[o2::phos::PHOSSimParams::Instance().mNLMMax];
+
+  int numberOfNotUnfolded = mClusters.size();
+
+  auto clu = mClusters.begin();
+
+  for (int i = 0; i < numberOfNotUnfolded; i++) {
+    if (clu->getNExMax() > -1) { //already unfolded
+      continue;
+    }
+    char nMultipl = clu->getMultiplicity();
+    char nMax = clu->getNumberOfLocalMax(maxAt, maxAtEnergy);
+    if (nMax > 1) {
+      unfoldOneCluster(&(*clu), nMax, maxAt, maxAtEnergy, digits);
+
+      clu = mClusters.erase(clu); //remove current and move iterator to next cluster
```

Comment:
vector::erase is very expensive procedure (requires copy for all elements after the erased one. If you cannot avoid clusters elimination and it happens often, you better fill 1st a temporary vector, flag entries which should be eliminated, then copy surviving ones to the final vector.